binfmt/libelf/libelf_coredump.c

```diff
@@ -423,9 +423,34 @@ static void elf_emit_memory(FAR struct elf_dumpinfo_s *cinfo, int memsegs)
 
   for (i = 0; i < memsegs; i++)
     {
-      elf_emit(cinfo, (FAR void *)cinfo->regions[i].start,
-               cinfo->regions[i].end -
-               cinfo->regions[i].start);
+      if (cinfo->regions[i].flags & PF_REGISTER)
```

Comment:
why not just copy unaligned data and emit aligned pointer ?

Reply:
Some chips require 4-byte register access, but you don't want to lose the memcpy optimization when copying memory.